### DISCUSSION TITLE
V3 Migration: React Automatic Runtime Error

### DISCUSSION THREAD

**Discussion by anaclumos** [facebook/docusaurus#9541] [v1-v2 migration]
```bash
Error: Docusaurus server-side rendering could not render static page with path /r/6B3AC8 because of error: React is not defined
    at render (main:621884:26)
    at async /Users/cho/Downloads/www/node_modules/p-map/index.js:57:22 {
  [cause]: ReferenceError: React is not defined
      at IconExternalLink (main:659484:5)
      at Uc (main:587364:44)
      at Xc (main:587366:253)
      at Z (main:587372:89)
      at Yc (main:587375:98)
      at $c (main:587374:140)
      at Z (main:587372:345)
      at Xc (main:587367:476)
      at Z (main:587372:89)
      at Yc (main:587375:98)
}
```

Getting the following error while upgrading to Docusaurus v3. Most likely related to the React Automatic Runtime. Any guides or document I should look into?

**Comment by Josh-Cena** [comment]
Does it work in local development? Could you try making a minimal reproduction? That path looks suspiciously like an autogenerated route—are you sure the code doesn't contain any hacks like relying on the `React` variable?

**Reply by anaclumos** [reply]
Ah, the path is simply because I hard-coded the `slug`. [It's this Repo](https://github.com/anaclumos/extracranial/blob/bbb2ad5ad6d7dcf4c9b9db8967fe0638a7c4a821/Research/pages/Exquisite%20Geometric%20Nature%20of%20the%20Universe.md?plain=1#L3)

I resolved this by re-adding `import React from 'react'` to where it was... Seems like React Automatic Runtime isn't working for some reason.
